### ISSUE TITLE
eXist v5: unable to load Replication Trigger

### ISSUE THREAD

**Issue by dizzzz** [eXist-db/messaging-replication#162]
to be investigated: in existdb v5 the startup triggers cannot be loaded from a XAR , see log.

note: the log is at database startup ; similar issue when XAR is freshly loaded in package manager.

```
2019-08-05 21:40:53,875 [main] INFO  (SyncTask.java [configure]:73) - Using DATA_DIR: /Users/xxxxxx/Development/GitHub/exist/exist-distribution/target/exist-distribution-5.0.0-RC9-SNAPSHOT-dir/etc/../data. Minimal disk space required for database to continue operations: 128mb 
2019-08-05 21:40:53,882 [main] INFO  (SyncTask.java [configure]:76) - Usable space on partition containing DATA_DIR: /Users/xxxxxx/Development/GitHub/exist/exist-distribution/target/exist-distribution-5.0.0-RC9-SNAPSHOT-dir/etc/../data: 146600mb 
2019-08-05 21:40:54,346 [main] INFO  (BlobStoreImplService.java [startSystem]:72) - Opened de-duplicating Blob Store v1. metadata=blob.dbx, store=blob/ 
2019-08-05 21:40:54,426 [main] WARN  (CollectionConfiguration.java [configureTrigger]:336) - Trigger class not found: org.exist.jms.replication.publish.ReplicationTrigger 
java.lang.ClassNotFoundException: org.exist.jms.replication.publish.ReplicationTrigger
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381) ~[?:1.8.0_162]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[?:1.8.0_162]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[?:1.8.0_162]
        at java.lang.Class.forName0(Native Method) ~[?:1.8.0_162]
        at java.lang.Class.forName(Class.java:264) ~[?:1.8.0_162]
        at org.exist.collections.CollectionConfiguration.configureTrigger(CollectionConfiguration.java:309) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfiguration.read(CollectionConfiguration.java:148) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfigurationManager.loadConfiguration(CollectionConfigurationManager.java:262) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfigurationManager.loadAllConfigurations(CollectionConfigurationManager.java:236) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfigurationManager.loadAllConfigurations(CollectionConfigurationManager.java:244) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfigurationManager.loadAllConfigurations(CollectionConfigurationManager.java:244) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfigurationManager.loadAllConfigurations(CollectionConfigurationManager.java:244) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfigurationManager.loadAllConfigurations(CollectionConfigurationManager.java:244) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfigurationManager.loadAllConfigurations(CollectionConfigurationManager.java:244) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfigurationManager.loadAllConfigurations(CollectionConfigurationManager.java:228) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.collections.CollectionConfigurationManager.startSystem(CollectionConfigurationManager.java:89) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.storage.BrokerPoolServicesManager.startSystemServices(BrokerPoolServicesManager.java:186) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.storage.BrokerPool._initialize(BrokerPool.java:604) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.storage.BrokerPool.initialize(BrokerPool.java:438) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.storage.BrokerPools.configure(BrokerPools.java:174) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.storage.BrokerPools.configure(BrokerPools.java:109) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.jetty.JettyStart.run(JettyStart.java:205) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.jetty.JettyStart.main(JettyStart.java:95) [exist-core-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_162]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_162]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_162]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_162]
        at org.exist.start.Main.invokeMain(Main.java:118) [exist-start-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.start.Main.runEx(Main.java:248) [exist-start-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.start.Main.run(Main.java:123) [exist-start-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at org.exist.start.Main.main(Main.java:60) [exist-start-5.0.0-RC9-SNAPSHOT.jar:5.0.0-RC9-SNAPSHOT]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_162]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_162]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_162]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_162]
        at org.codehaus.mojo.appassembler.booter.AppassemblerBooter.executeMain(AppassemblerBooter.java:257) [appassembler-booter-2.1.0.jar:?]
        at org.codehaus.mojo.appassembler.booter.AppassemblerBooter.main(AppassemblerBooter.java:77) [appassembler-booter-2.1.0.jar:?]
2019-08-05 21:40:54,532 [main] INFO  (NativeBroker.java [sync]:3654) - Memory: 196,096K total; 2,796,544K max; 128,807K free 
2019-08-05 21:40:54,537 [main] INFO  (BTree.java [printStatistics]:2665) - dom.dbx INDEX Buffers occupation : 20% (13 out of 64) Cache efficiency : 100% 
2019-08-05 21:40:54,539 [main] INFO  (DOMFile.java [printStatistics]:1424) - dom.dbx DATA Buffers occupation : 7% (19 out of 256) Cache efficiency : 99% 
2019-08-05 21:40:54,541 [main] INFO  (BTree.java [printStatistics]:2665) - collections.dbx INDEX Buffers occupation : 16% (10 out of 64) Cache efficiency : 88% 
2019-08-05 21:40:54,542 [main] INFO  (BFile.java [printStatistics]:409) - collections.dbx DATA Buffers occupation : 12% (8 out of 64) Cache efficiency : 80% 
2019-08-05 21:40:54,542 [main] INFO  (BTree.java [printStatistics]:2665) - values.dbx INDEX Buffers occupation : 0% (0 out of 64) Cache efficiency : N/A 
2019-08-05 21:40:54,542 [main] INFO  (BFile.java [printStatistics]:409) - values.dbx DATA Buffers occupation : 0% (0 out of 64) Cache efficiency : N/A 
2019-08-05 21:40:54,734 [main] INFO  (BouncyCastleJceProviderStartupTrigger.java [execute]:48) - Registered JCE Security Provider: org.bouncycastle.jce.provider.BouncyCastleProvider 
```

**Comment by dizzzz** [comment]
2019-08-05 20:56:02,760 [qtp1488608048-29] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,455 [qtp1488608048-30] INFO  (Deployment.java [installAndDeploy]:219) - Package http://exist-db.org/messaging-replication depends on http://exist-db.org/apps/shared 
2019-08-05 20:56:56,456 [qtp1488608048-30] DEBUG (Deployment.java [installAndDeploy]:222) - Package http://exist-db.org/apps/shared already installed 
2019-08-05 20:56:56,456 [qtp1488608048-30] DEBUG (Deployment.java [installAndDeploy]:243) - Package http://exist-db.org/apps/shared already installed 
2019-08-05 20:56:56,457 [qtp1488608048-30] INFO  (Deployment.java [installAndDeploy]:267) - Installing package /db/system/repo/messaging-replication-1.0.0-SNAPSHOT.xar 
2019-08-05 20:56:56,620 [qtp1488608048-30] TRACE (Logger.java [trace]:33) - Content dir ''{0}'' (exists:{1}/isdir:{2}), and ''{3}'' (exists:{4}/isdir:{5}) 
2019-08-05 20:56:56,644 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,645 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,645 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - File '/Users/wessels/Development/GitHub/exist/exist-distribution/target/exist-distribution-5.0.0-RC9-SNAPSHOT-dir/etc/../data/expathrepo/jms-1.0.0-SNAPSHOT/.exist/classpath.txt' does not exist 
2019-08-05 20:56:56,646 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,647 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,647 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,647 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,648 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,648 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,649 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,649 [qtp1488608048-30] DEBUG (Logger.java [debug]:38) - Trying to resolve ''{0}'' within ''{1}'' 
2019-08-05 20:56:56,656 [qtp1488608048-30] INFO  (Deployment.java [installAndDeploy]:280) - Deploying package http://exist-db.org/messaging-replication 
2019-08-05 20:56:57,090 [qtp1488608048-24] DEBUG (Logger.java [debug]:38) - Package ''{0}'', resolve in {1}: ''{2}'' ({3}) 
2019-08-05 20:56:57,091 [qtp1488608048-24] DEBUG (Logger.java [debug]:38) - Package ''{0}'', no URI in {1} 
2019-08-05 20:56:57,091 [qtp1488608048-24] DEBUG (Logger.java [debug]:38) - Package ''{0}'', resolve in {1}: ''{2}'' ({3}) 
2019-08-05 20:56:57,091 [qtp1488608048-24] DEBUG (Logger.java [debug]:38) - Package ''{0}'', not in {1}: ''{2}'' 
2019-08-05 20:56:57,091 [qtp1488608048-24] DEBUG (Logger.java [debug]:38) - Package ''{0}'', resolve in {1}: ''{2}'' ({3})

**Comment by dizzzz** [comment]
fixed by #166
